umbraco-cloud/set-up/project-settings/umbraco-cicd/SampleCiCdPipeline.md

```diff
@@ -0,0 +1,206 @@
+
+1. **Load from Existing Repositories**
+
+    Begin by loading your existing repositories into Azure DevOps.
+
+    ![Pipeline1.png](../../images/Pipeline1.png)
+
+2. **Select the Repository**
+
+    Choose the repository you want to use for the pipeline.
+
+    ![Pipeline2.png](../../images/Pipeline2.png)
+
+3. **Configure the Pipeline**
+
+    Next, configure the pipeline to use the existing YAML file from your selected repository.
+
+    ![Pipeline3.png](../../images/Pipeline3.png)
+
+4. **Select the YAML File from the Repository**
+
+    Finally, specify the YAML file that defines your pipeline. You can find a YAML file for Azure DevOps and Google Actions, respectively here [link to the “Sample Script” section]
+
+    ![Pipeline4.png](../../images/Pipeline4.png)
+
+By following these steps, you'll have successfully set up a CI/CD pipeline in Azure DevOps for your Umbraco Cloud project.
+
+## Configuring a Staged Pipeline in Azure DevOps
+
+This guide outlines how to set up a multi-stage pipeline in Azure DevOps using the provided sample scripts. The pipeline is defined in an Azure YAML file and consists of three key stages:
+
+* **Preflight**: Validates if there are any remote changes since the last successful deployment.
+* **Build, Test, and Package**: Executes standard build and test procedures, and packages the project.
+* **Deploy**: Manages the creation, initiation, and monitoring of the actual deployment.
+
+### Import Scripts and Configure Variables
+
+1. **Copy Scripts**: Start by copying the pipeline and associated scripts into a new folder within your own project repository.
+
+2. **Configure Variables**: Open the `azure-release-pipeline.yaml` file and set the appropriate variables in the variables section. These variables can be found in the Settings -> Advanced section of your project on the [Umbraco Cloud Portal](https://www.s1.umbraco.io/projects).
+
+3. **Variable Requirements**: The sample pipeline expects the following variables to be set:
+    * `umbraco-cloud-api-key`: The API key needed to access the Umbraco Cloud API.
+    * `git-pat`: A Personal Access Token (PAT) for Git interactions.
+    * `project-id`: The ID of the project that the pipeline will be used for.
+
+By following these steps, you'll have a staged pipeline configured in Azure DevOps tailored for your Umbraco Cloud project.
+
+### Pipeline Stage: Preflight Checks
+
+This stage involves making an API call to retrieve the latest completed deployment. The goal is to identify any remote changes that may have occurred since the last successful deployment. If such changes are detected, they are applied to a new Git branch and pushed. A manual validation gate is introduced if changes are found.
+
+Note: The current pipeline script has room for improvement. For instance, it may falsely indicate remote changes even after a patch has been applied. It also requires manual intervention twice, as changes persist until they are deployed.
```

Comment:
```suggestion
{% hint style="info" %}

The current pipeline script has room for improvement. For instance, it may falsely indicate remote changes even after a patch has been applied. It also requires manual intervention twice, as changes persist until they are deployed.

{% endhint %}

```